src/Randomizer.CrossPlatform/Services/TrackerWindowService.cs

```diff
@@ -43,16 +157,318 @@ public void StartTracker()
+        _medallions.Clear();
+
+        var models = new List<TrackerWindowPanelViewModel>();
+
+        foreach (var spot in layout.GridLocations)
+        {
+            var panel = GetPanelViewModel(spot);
+            if (panel != null)
+            {
+                models.Add(panel);
+            }
+        }
+
+        _model.Panels = models;
+
+        ITaskService.Run(() =>
+        {
+            Options.GeneralOptions.SelectedLayout = layout.Name;
+            Options.Save();
+        });
+    }
+
+    public void ToggleTimer()
+    {
+        tracker.ToggleTimer();
     }
 
-    private MessageWindow DisplayError(string message) => OpenMessageWindow(message);
+    public void ResetTimer()
+    {
+        tracker.ResetTimer();
+    }
+
+    public void SetConnector(SnesConnectorType snesConnectorType)
+    {
+        tracker.AutoTracker?.SetConnector(Options.GeneralOptions.SnesConnectorSettings, snesConnectorType);
+    }
+
+    public void OpenAutoTrackerFolder()
+    {
+        var path = Options.AutoTrackerScriptsOutputPath;
+        if (string.IsNullOrEmpty(path))
+        {
+            path = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "SMZ3CasRandomizer", "AutoTrackerScripts");
+        }
+
+        CrossPlatformTools.OpenDirectory(path);
+    }
+
+    private TrackerWindowPanelViewModel? GetPanelViewModel(UIGridLocation? gridLocation)
+    {
+        return gridLocation?.Type switch
+        {
+            UIGridLocationType.Items => GetItemPanelViewModel(gridLocation),
+            UIGridLocationType.Dungeon => GetDungeonPanelViewModel(gridLocation),
+            UIGridLocationType.SMBoss => GetBossPanelViewModel(gridLocation),
+            _ => null
+        };
+    }
+
+    private TrackerWindowPanelViewModel GetItemPanelViewModel(UIGridLocation gridLocation)
+    {
+        var items = new Dictionary<Item, string>();
+        List<Item>? connectedItems = null;
+
+        var labelImage = uiService.GetSpritePath("Items", gridLocation.Image ?? "", out _);
+
+        foreach (var itemName in gridLocation.Identifiers)
+        {
+            var item = itemService.FirstOrDefault(itemName);
+            if (item == null)
+            {
+                logger.LogError("Item {ItemName} could not be found", itemName);
+                continue;
+            }
+
+            if (item.Type == ItemType.Bottle)
+            {
+                connectedItems = itemService.LocalPlayersItems()
+                    .Where(x => x.Type.IsInCategory(ItemCategory.Bottle)).ToList();
+            }
+
+            var fileName = uiService.GetSpritePath(item);
+            if (fileName == null)
+            {
+                logger.LogError("Image for {ItemName} could not be found", item.Name);
+                continue;
+            }
+
+            items[item] = fileName;
+        }
+
+        if (gridLocation.Identifiers.First() == "Power Bomb")
+        {
+            var a = "1";
```

Comment:
GitHub is flagging this block because the variable is unused. Leftover debugging code?

Reply:
Nope, that's clearly load bearing code. The entire application topples down without that a = "1" line.